src/app/pages/back-to-office/__tests__/back-to-office.page.spec.ts

```diff
@@ -1,31 +1,170 @@
+                      code: '',
+                      description: '',
+                      outcome: '',
+                    },
+                  },
+                  eyesightTest: {},
+                },
+                activityCode: '28',
+                journalData: {
+                  candidate: {
+                    candidateName: 'Joe Bloggs',
+                    driverNumber: '123',
+                  },
+                },
+                rekey: false,
+              },
+            },
+          }),
+        }),
+      ],
       providers: [
-        { provide: NavController, useClass: NavMock },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        { provide: AuthenticationProvider, useClass: AuthenticationProviderMock },
+        { provide: DateTimeProvider, useClass: DateTimeProviderMock },
+        { provide: ScreenOrientation, useClass: ScreenOrientationMock },
+        { provide: Insomnia, useClass: InsomniaMock },
+        { provide: DeviceProvider, useClass: DeviceProviderMock },
         { provide: RouteByCategoryProvider, useClass: RouteByCategoryProviderMock },
       ],
-    }).compileComponents();
+    });
+  });
 
+  beforeEach(waitForAsync(() => {
     fixture = TestBed.createComponent(BackToOfficePage);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    screenOrientation = TestBed.inject(ScreenOrientation);
+    insomnia = TestBed.inject(Insomnia);
+    deviceProvider = TestBed.inject(DeviceProvider);
+    store$ = TestBed.inject(Store);
+    router = TestBed.inject(Router);
+    spyOn(store$, 'dispatch');
+    spyOn(router, 'navigate');
+    spyOn(router, 'navigate').and.returnValue(Promise.resolve(true));
   }));
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  describe('Class', () => {
+    describe('ionViewDidEnter', () => {
+      it('should disable test inhibitions when not in practice mode', async (done) => {
+        component.isPracticeMode = true;
+        await component.ionViewDidEnter();
+        expect(deviceProvider.disableSingleAppMode).not.toHaveBeenCalled();
+        expect(screenOrientation.unlock).toHaveBeenCalled();
+        expect(insomnia.allowSleepAgain).toHaveBeenCalled();
+        done();
+      });
+      it('should disable test inhibitions when not in practice mode', async (done) => {
+        component.isPracticeMode = false;
+        await component.ionViewDidEnter();
+        expect(deviceProvider.disableSingleAppMode).toHaveBeenCalled();
+        expect(screenOrientation.unlock).toHaveBeenCalled();
+        expect(insomnia.allowSleepAgain).toHaveBeenCalled();
+        done();
+      });
+    });
+  });
+
+  describe('goToJournal', () => {
+    it('should call the popTo method in the navcontroller if not in practice mode', () => {
```

Comment:
can we amend the description here to be something like
'should navigate to the journal page when not in practice mode'

and possibly be more verbose in the expectations? e.g.
```
expect(store$.dispatch).toHaveBeenCalledWith(DeferWriteUp());
expect(router.navigate).toHaveBeenCalledWith([JOURNAL_PAGE], { replaceUrl: true });
```
Also, I can see the comment below re MES-6867, but the code is already implemented here so would it make sense to write a small unit test - will leave that down to you to decide

Reply:
yes fair point have updated to include unit test